llvm/lib/Target/RISCV/Disassembler/RISCVDisassembler.cpp

```diff
@@ -257,15 +248,12 @@ static DecodeStatus decodeVMaskReg(MCInst &Inst, uint64_t RegNo,
                                    uint64_t Address,
                                    const MCDisassembler *Decoder) {
   MCRegister Reg = RISCV::NoRegister;
-  switch (RegNo) {
-  default:
+  if (RegNo > 2) {
     return MCDisassembler::Fail;
-  case 0:
+  } else if (RegNo == 0) {
```

Comment:
No else after `if` that returns.

Reply:
Because there is an initial default value in front, I omitted the "else" part.

Reply:
I meant.

```
if (RegNo > 2)
  return MCDisassembler::Fail;
if (RegNo == 0) <- no need for else before this.
  Reg = RISCV::V0
```

Though reading the code again. I think this is simpler

```
if (RegNo > 2)
  return MCDisassembler::Fail;
MCRegister Reg = (RegNo == 0) ? RISCV::V0 : RISCV::NoRegister;
```

Reply:
that's great!

Reply:
> I meant.
> 
> ```
> if (RegNo > 2)
>   return MCDisassembler::Fail;
> if (RegNo == 0) <- no need for else before this.
>   Reg = RISCV::V0
> ```
> 
> Though reading the code again. I think this is simpler
> 
> ```
> if (RegNo > 2)
>   return MCDisassembler::Fail;
> MCRegister Reg = (RegNo == 0) ? RISCV::V0 : RISCV::NoRegister;
> ```

that's great!